javagen/src/main/java/com/azure/autorest/template/ClientMethodTemplateBase.java

```diff
@@ -76,7 +77,35 @@ protected static void generateProtocolMethodJavadoc(ClientMethod clientMethod, J
             // Response body
             IType responseBodyType;
             if (JavaSettings.getInstance().isDataPlaneClient()) {
-                responseBodyType = clientMethod.getProxyMethod().getRawResponseBodyType();
+                // special handling for paging method
+                if (clientMethod.getType() == ClientMethodType.PagingSync || clientMethod.getType() == ClientMethodType.PagingAsync || clientMethod.getType() == ClientMethodType.PagingAsyncSinglePage || clientMethod.getType() == ClientMethodType.PagingSyncSinglePage) {
+                    String itemName = clientMethod.getMethodPageDetails().getItemName();
+                    // rawResponseType has properties: 'value' and 'nextLink'
+                    IType rawResponseType = clientMethod.getProxyMethod().getRawResponseBodyType();
+                    if (!(rawResponseType instanceof ClassType)) {
+                        throw new IllegalArgumentException("clientMethod.getProxyMethod().getRawResponseBodyType() should be ClassType for paging method.");
+                    }
+                    ClientModel model = ClientModelUtil.getClientModel(((ClassType) rawResponseType).getName());
+                    List<ClientModelProperty> properties = new ArrayList<>();
+                    traverseProperties(model, properties);
+                    responseBodyType = properties.stream()
+                            .filter(property -> property.getName().equals(itemName))
+                            .map(clientModelProperty -> clientModelProperty.getClientType())
+                            .map(valueListType -> {
+                                // value type is List<T>, we need to get the typeArguments
+                                if (!(valueListType instanceof ListType)) {
+                                    throw new IllegalArgumentException("value type must be list");
```

Comment:
similar to your text in above exception, mention "paging method" so it is easier to diagnose.

mind the sentence and proper punctuation (I assume I don't need to say it in every PR?).

Reply:
Please also include the `IType.toString()` of the rawResponseType, for diagnostic in all exceptions.